src/App.js

```diff
@@ -88,31 +90,43 @@ class App extends React.PureComponent<Props, State> {
           href="https://fonts.googleapis.com/css?family=Roboto"
           rel="stylesheet"
         />
-        <I18nextProvider i18n={this.i18n}>
-          <LanguageContext.Provider value={this.props.language}>
-            <CloseContext.Provider value={this.props.onClose}>
-              <UserContext.Provider
-                value={{
-                  user: this.props.user,
-                  onLogin: this.props.onLogin,
-                  onSocialLogin: this.props.onSocialLogin,
-                  onLogout: this.props.onLogout,
-                  loginToken: this.props.loginToken,
-                }}
-              >
-                <BookingStateProvider>
-                  <SearchStateProvider>
-                    <SelectUrlBooking
-                      wasSelected={this.state.urlBookingWasSelected}
-                      setSelected={this.urlBookingSelected}
-                    />
-                    <Routes initialRoute={this.props.initialRoute} />
-                  </SearchStateProvider>
-                </BookingStateProvider>
-              </UserContext.Provider>
-            </CloseContext.Provider>
-          </LanguageContext.Provider>
-        </I18nextProvider>
+        <ErrorBoundaryProvider>
```

Comment:
why exactly is done using Context API, original example is much simpler:

https://reactjs.org/docs/error-boundaries.html#introducing-error-boundaries

```
if (this.state.hasError) {
  return <ErrorMessage />;
}

return  this.props.children;
```
and then:
```
<ErrorBoundary>
  <App {...props} />
</ErrorBoundary>
```

Reply:
I thought to make it more "flexible" 😅 But I agree its a pre mature optimization...